DNN Platform/Library/Data/SqlDatabaseConnectionProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿#region Copyright
+// 
+// DotNetNuke® - http://www.dotnetnuke.com
+// Copyright (c) 2002-2014
+// by DotNetNuke Corporation
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
+// of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+// DEALINGS IN THE SOFTWARE.
+#endregion
+
+#region usings
+
+using System;
+using System.Linq;
+
+#endregion
+
+namespace DotNetNuke.Data
+{
+    using System.Data;
+    using System.Data.SqlClient;
+
+    using DotNetNuke.Data.PetaPoco;
+
+    using Microsoft.ApplicationBlocks.Data;
+
+    public class SqlDatabaseConnectionProvider : DatabaseConnectionProvider
+    {
+        public override void ExecuteCommand()
+        {
+            //Create a new connection
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                //Create a new command (with no timeout)
+                var command = new SqlCommand(Query, connection) { CommandTimeout = 0 };
```

Comment:
Is this supposed to override/ignore the `CommandTimeout` property?  The [calling code](#diff-ac4f83a40d8acd75b4980dfa00e3d574R115) sets `CommandTimeout` to `0`, should this be done from the calling code, or always set here in the provider?  Off the top of my head, I would assume the calling code should be responsible for setting the timeout.